specification/app/resource-manager/Microsoft.App/stable/2025-01-01/ConnectedEnvironmentsStorages.json

```diff
@@ -0,0 +1,290 @@
+          },
+          "default": {
+            "description": "Common error response.",
+            "schema": {
+              "$ref": "./CommonDefinitions.json#/definitions/DefaultErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Create or update environments storage": {
+            "$ref": "./examples/ConnectedEnvironmentsStorages_CreateOrUpdate.json"
+          }
+        }
+      },
+      "delete": {
+        "tags": [
+          "ConnectedEnvironmentsStorages"
+        ],
+        "summary": "Delete storage for a connectedEnvironment.",
+        "description": "Delete storage for a connectedEnvironment.",
+        "operationId": "ConnectedEnvironmentsStorages_Delete",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "name": "connectedEnvironmentName",
+            "in": "path",
+            "description": "Name of the Environment.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "storageName",
+            "in": "path",
+            "description": "Name of the storage.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK"
+          },
+          "204": {
+            "description": "Storage does not exist"
+          },
+          "default": {
+            "description": "Common error response.",
+            "schema": {
+              "$ref": "./CommonDefinitions.json#/definitions/DefaultErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "List environments storages by subscription": {
+            "$ref": "./examples/ConnectedEnvironmentsStorages_Delete.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "ConnectedEnvironmentStoragesCollection": {
+      "description": "Collection of Storage for Environments",
+      "required": [
+        "value"
+      ],
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "Collection of storage resources.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ConnectedEnvironmentStorage"
+          }
+        }
+      }
+    },
+    "ConnectedEnvironmentStorage": {
+      "description": "Storage resource for connectedEnvironment.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v5/types.json#/definitions/ProxyResource"
```

Comment:
>nt/v5/types.jso [](http://example.com/codeflow?start=64&length=15)

pl dont mix and match the versions in the same json file. This will cause issues downstream with SDK generation etc. Revert to v3 everywhere

Reply:
![image](https://github.com/user-attachments/assets/54c897b1-64a0-4f47-94ed-dceb63d2963c)

Reply:
Breaking change reviewer also approval the change from v3 to v5
 https://msazure.visualstudio.com/One/_workitems/edit/31699358

Reply:
And we also use v5 in previous api version (with ARM Approval) https://github.com/Azure/azure-rest-api-specs/pull/31545

Reply:
@rkmanda Roopesh Manda would you review again? it's already using v5 in release-app-Microsoft.App-2024-10-02-preview(https://github.com/Azure/azure-rest-api-specs/pull/31545), We want to keep v5 in  2025-01-01 too, avoiding breaking change